app/src/apis/memberApi.ts

```diff
@@ -56,9 +60,11 @@ export class FetchMemberAPI implements MemberAPI {
       if (!response.ok) {
         throw new Error(`HTTP error! Status: ${response.status}`)
       }
-    } catch (error) {
+      show(ToastType.Success, 'Updated Member Details')
+    } catch (error: any) {
+      show(ToastType.Warning, error.message)
       console.error('Error:', error)
-      return
+      throw error
     }
```

Comment:
I feel you're violating Single Responsibility here, you're making API calls, displaying errors to the UI and handling errors at the same. You should just throw an error and have some lower level code somewhere decide what to do with the error. The way you're doing it forces you to go and implement this logic in all your API code instead of just where the API classes are used.

Reply:
The show function is from useToastStore (not a Vue function) and it keeps UI interactions separate from component logic, and although it might seem like error handling occurs within the component, show is still a lower-level code like you mentioned, fetched from the centralized Pinia store.

Reply:
Yes, you have a `try..catch` block around your code which is error handling. Instead you would have that where you are "using" `FetchMemberAPI` which is most likely going to be a composable soon although, right now you're calling it directly in a view.